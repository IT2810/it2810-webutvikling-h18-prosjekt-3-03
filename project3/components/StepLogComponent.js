import React, { Component } from 'react';
import { FlatList, StyleSheet, Text, View, Image } from 'react-native';
//<div>Icons made by <a href="http://www.freepik.com" title="Freepik">Freepik</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a> is licensed by <a href="http://creativecommons.org/licenses/by/3.0/" title="Creative Commons BY 3.0" target="_blank">CC 3.0 BY</a></div>
// <div>Icons made by <a href="https://www.flaticon.com/authors/nikita-golubev" title="Nikita Golubev">Nikita Golubev</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a> is licensed by <a href="http://creativecommons.org/licenses/by/3.0/" title="Creative Commons BY 3.0" target="_blank">CC 3.0 BY</a></div>


export default class FlatListBasics extends Component {
  renderListSeparator = () => {
     return (
       <View
         style={styles.itemSeperator}
       />
     );
   };

   renderListHeader = () => {
      return (
        <View style={styles.listHeader}>
        <Text style={styles.headerText}> Log </ Text></ View>
      );
    };


  render() {
    return (
      <View style={styles.container}>
        <FlatList
        ItemSeparatorComponent={this.renderListSeparator}
        ListHeaderComponent={this.renderListHeader}
          data={[
            {key: new Date("2018","09","06").toDateString(), steps:3000, goal: 5000, achieved: false},
            {key: new Date("2018","09","07").toDateString(), steps:2000, goal: 3000, achieved: false},
            {key: new Date("2018","09","08").toDateString(), steps:7000, goal: 7000, achieved: true},
            {key: new Date("2018","09","09").toDateString(), steps:13000, goal: 10000, achieved: true},
          ]}
          renderItem={({item}) => <View style={styles.container} ><View style={styles.cont2}><Text style={styles.dateText}>{item.key}</Text>
          <Image style={styles.stepImg} source={ require('../assets/run.png')} /><Text style={styles.stepText}>{item.steps}/{item.goal}</Text></ View>
          {item.achieved ? <Image style={styles.starImg} source={ require('../assets/favourites.png')} /> : <Image style={styles.starImg} source={ require('../assets/empty.png')} />}</View>}

        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
   flex: 1,
  alignItems: 'flex-start',
  flexDirection: 'row',
  flexWrap: 'wrap',
  },
  dateText: {
    paddingTop: 4,
    margin: "4%",
    marginRight:0,
    fontSize: 18,
    flexDirection: 'row',
    width: "48%",
  },
  stepText: {
    paddingTop: 6,
    margin: "4%",
    marginLeft: "2%",
    fontSize: 14,
    flexDirection: 'row',
  },
  starImg:{
    margin: "4%",
  },
  stepImg:{
    marginTop: "5%",
  },
  itemSeperator:{
    height: 1,
    width: "96%",
    backgroundColor: "#CED0CE",
    marginLeft: "2%"
  },
  cont2: {
   flex: 1,
  alignItems: 'flex-start',
  flexDirection: 'row',
  flexWrap: 'wrap',
  },
  listHeader:{
    width: "100%",
    height: 40,
    backgroundColor: "#555555",
    alignItems:"center",

    marginTop: "8%",
  },
  headerText:{
    color: "#FFFFFF",
    fontSize: 20,
    marginTop: 5,
  }
})
