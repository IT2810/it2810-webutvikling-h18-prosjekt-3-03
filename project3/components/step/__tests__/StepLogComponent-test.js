import React from 'react';
import {View} from 'react-native';
import renderer from 'react-test-renderer';
import StepLogComponent from '../StepLogComponent';

it('renders correctly', () => {
    const tree = renderer
        .create(<View>
            <StepLogComponent
                data={[]}
            />
            <StepLogComponent
                data={[{key: new Date(0).toDateString(), steps: 0, goal: 10000, achieved: false}]}
            />
            <StepLogComponent
                data={[{key: new Date(1).toDateString(), steps: 10001, goal: 10000, achieved: true}]}
            />
            <StepLogComponent
                data={[
                    {key: new Date(2).toDateString(), steps: 2, goal: 10000, achieved: false},
                    {key: new Date(3).toDateString(), steps: 10003, goal: 10000, achieved: true},
                ]}
            />
        </View>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
