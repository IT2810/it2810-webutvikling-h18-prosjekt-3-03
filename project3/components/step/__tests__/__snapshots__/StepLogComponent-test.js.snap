// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  />
</View>
`;
